src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -436,4 +435,32 @@ IEnumerator IEnumerable.GetEnumerator()
             }
         }
     }
+
+    private class RequestActivityData
+    {
+        [DynamicDependency(DynamicallyAccessedMemberTypes.PublicProperties, typeof(HttpContext))]
+        public RequestActivityData(long timeStamp, HttpContext httpContext)
+        {
+            TimeStamp = timeStamp;
+            HttpContext = httpContext;
+        }
+
+        long TimeStamp { get; }
+        HttpContext HttpContext { get; }
+    }
+
+    private class RequestExceptionActivityData
+    {
+        [DynamicDependency(DynamicallyAccessedMemberTypes.PublicProperties, typeof(HttpContext))]
```

Comment:
What about Exception.Message? I imagine people would want to get that in a trace. In the runtime we also preserve StackTrace:

[https://github.com/dotnet/runtime/blob/5fa6dd364982be4ffd83358adbf130d88049c72a/src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs#L260-L261](https://github.com/dotnet/runtime/blob/5fa6dd364982be4ffd83358adbf130d88049c72a/src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs#L260-L261)